src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas2/Cas2StatusUpdateTest.kt

```diff
@@ -212,41 +214,45 @@ class Cas2StatusUpdateTest(
               .expectStatus()
               .isCreated
 
-            Assertions.assertThat(realStatusUpdateRepository.count()).isEqualTo(1)
-            Assertions.assertThat(realStatusUpdateDetailRepository.count()).isEqualTo(1)
+            assertThat(realStatusUpdateRepository.count()).isEqualTo(1)
+            assertThat(realStatusUpdateDetailRepository.count()).isEqualTo(1)
 
             val persistedStatusUpdate =
               realStatusUpdateRepository.findFirstByApplicationIdOrderByCreatedAtDesc(application.id)
-            Assertions.assertThat(persistedStatusUpdate!!.assessment!!.id).isEqualTo(assessmentId)
+            assertThat(persistedStatusUpdate!!.assessment!!.id).isEqualTo(assessmentId)
 
             val persistedStatusDetailUpdate =
               realStatusUpdateDetailRepository.findFirstByStatusUpdateIdOrderByCreatedAtDesc(persistedStatusUpdate!!.id)
-            Assertions.assertThat(persistedStatusDetailUpdate).isNotNull
+            assertThat(persistedStatusDetailUpdate).isNotNull
 
             val appliedStatus = Cas2ApplicationStatusSeeding.statusList()
               .find { status ->
                 status.id == persistedStatusUpdate.statusId
               }
 
-            Assertions.assertThat(appliedStatus!!.name).isEqualTo("offerDeclined")
-            Assertions.assertThat(appliedStatus.statusDetails?.find { detail -> detail.id == persistedStatusDetailUpdate?.statusDetailId })
+            assertThat(appliedStatus!!.name).isEqualTo("offerDeclined")
+            assertThat(appliedStatus.statusDetails?.find { detail -> detail.id == persistedStatusDetailUpdate?.statusDetailId })
               .isNotNull()
 
             emailAsserter.assertEmailsRequestedCount(1)
-            emailAsserter.assertEmailRequested(
+            val email = emailAsserter.assertEmailRequested(
               toEmailAddress = applicant.email!!,
               templateId = "ef4dc5e3-b1f1-4448-a545-7a936c50fc3a",
               personalisation = mapOf(
                 "applicationStatus" to "Offer declined or withdrawn",
-                "dateStatusChanged" to submittedAt.toLocalDate().toCas2UiFormat(), // This needs to be something else
-                "timeStatusChanged" to submittedAt.toCas2UiFormattedHourOfDay(),
+                "dateStatusChanged" to submittedAt.toLocalDate().toCas2UiFormat(),
                 "nomsNumber" to "123NOMS",
                 "applicationType" to "Home Detention Curfew (HDC)",
                 "applicationUrl" to applicationOverviewUrlTemplate.replace("#id", application.id.toString()),
               ),
               replyToEmailId = notifyConfig.emailAddresses.cas2ReplyToId,
             )
 
+            assertThat(email.personalisation["timeStatusChanged"] as String).isIn(
+              requestMadeAt.toCas2UiFormattedHourOfDay(),
```

Comment:
Is this checking the time on the email against two timestamps - one made at the start of the test, and one at the point that this assertion takes place? So if the test was really lagging it might have happened minutes away from  the second `.now()` as well? 

It looks like its quite easy to mock OffsetDateTime.now(), if I understand what Dan has done here, so I wonder if it would be worth doing that?  https://github.com/ministryofjustice/hmpps-approved-premises-api/blob/d4640a3f832030380bc42307871fc79d2a7806e9/src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas2/Cas2ApplicationTest.kt#L246

Reply:
That's correct, when it failed for me I suspect the test started on the 59th second of a minute and ended in the 0th second of the next minute. Whilst i'd be surprised if the test took > 60 seconds to run, I agree that what i've done here isn't water tight and not particularly readable either.

I agree mocking statics is a better and more readable solution so i've updated the commit to do that instead. Ideally we'd use a 'clock' in the code that we can then mutate during integration tests but that's a bit more effort to setup.

I had to introduce a try/finally block to make sure we tidy the static mock up, sorry if the diff is a bit hard to read!